### PR TITLE
Absence: Prevent duplicate Absences

### LINKED ISSUE #55: Absences: skip absence creation if the student already has an entry for the date and lesson

### PR DESCRIPTION
**Pull request by jonaslanglotz**
Resolves #55 by creating a unique index for the fields `studentId`, `lessonIndex` and `date` on the absence entity. It also adds a new error type to the absenceCreate endpoint.

### CODE REVIEW

**Review by EricZimmer1 — APPROVED** [review]
Looks good to me

File: server/src/entity/Absence.ts [hunk 1]
```diff
@@ -45,6 +46,9 @@ export class Absence extends BaseEntity {
     @Field()
     exam: boolean
 
+    @Column()
+    studentId: string
```
Comment (bucherfa) [inline server/src/entity/Absence.ts:50]: Where is the studentId set? Shouldn't it be set in the create mutation or is there an automation?

**Review by bucherfa — CHANGES_REQUESTED** [review]
just the one feedback

File: server/src/entity/Absence.ts (hunk 1 above)
Comment (jonaslanglotz) [reply server/src/entity/Absence.ts:50]: TypeORM will automagically assign this value with the foreign key belonging to the student relation and will keep it in sync accordingly.

**Review by bucherfa — APPROVED** [review]

**Pull request merged**